kevm-pyk/src/kevm_pyk/kevm.py

```diff
@@ -644,3 +663,41 @@ def kevm_node_printer(kevm: KEVM, proof: APRProof) -> NodePrinter:
     if type(proof) is APRProof:
         return KEVMAPRNodePrinter(kevm, proof)
     raise ValueError(f'Cannot build NodePrinter for proof type: {type(proof)}')
+
+
+def compute_jumpdests(sections: list[KInner]) -> KInner:
+    """Analyzes a list of KInner objects representing sections of bytecode to compute jump destinations.
+
+    :param sections: A section is expected to be either a KToken of sort Bytes or a KApply with the label of '#buf(_,_)_BUF-SYNTAX_Bytes_Int_Int'.
+    :type sections: list[KInner]
+    :return: This function iterates over each section, appending the jump destinations (0x5B) from the bytecode in a KAst Set.
+    :rtype: KInner
+    """
+    offset = 0
+    jumpdests = []
+    for s in sections:
+        if type(s) is KApply and s.label == KLabel('#buf(_,_)_BUF-SYNTAX_Bytes_Int_Int'):
+            width_token = s.args[0]
+            assert type(width_token) is KToken
```

Comment:
Could we also assert here that the token is of sort `Int` or does that hold by construction?

Reply:
I'd say that it holds by construction of `#buf`. I'm not sure how possible is to have a symbolic length, but in this case, `args[0]` would be a `KVariable` instead of a `KToken`.